File: go/62.go
package main

func uniquePaths(m int, n int) int {
	ret := float64(1.0)

  n_f := float64(n-1)
  m_f := float64(m-1)
	for i := float64(1); i <= m_f; i++ {
		ret *= float64(n_f + i)
    ret /= i
	}

	return int(ret)
}

File: cpp/62.cpp
#include <iostream>

using namespace std;

class Solution {
public:
  int uniquePaths(int m, int n) {
    double ret = 1;

    --m, --n;
    int all = m + n;
    for (int i = 1; i <= n; i++)
      ret *= i, ret /= i;
    for (int i = 1; i <= m; i++)
      ret *= (n + i), ret /= i;

    return static_cast<int>(ret);
  }
};
